fsspec/dircache.py

```diff
@@ -93,6 +97,95 @@ def __iter__(self):
 
     def __reduce__(self):
         return (
-            DirCache,
+            MemDirCache,
             (self.use_listings_cache, self.listings_expiry_time, self.max_paths),
         )
+
+
+class FileDirCache(MutableMapping):
+    def __init__(
+        self,
+        use_listings_cache=True,
+        listings_expiry_time=None,
+        listings_cache_location=None,
+        **kwargs,
+    ):
+        """
+
+        Parameters
+        ----------
+        use_listings_cache: bool
+            If False, this cache never returns items, but always reports KeyError,
+            and setting items has no effect
+        listings_expiry_time: int or float (optional)
+            Time in seconds that a listing is considered valid. If None,
+            listings do not expire.
+        listings_cache_location: str (optional)
+            Directory path at which the listings cache file is stored. If None,
+            an autogenerated path at the user folder is created.
+
+        """
+        import platformdirs
+        from diskcache import Cache
+
+        listings_expiry_time = listings_expiry_time and float(listings_expiry_time)
+
+        if listings_cache_location:
+            listings_cache_location = Path(listings_cache_location) / str(
+                listings_expiry_time
+            )
+            listings_cache_location.mkdir(exist_ok=True, parents=True)
+        else:
+            listings_cache_location = Path(
+                platformdirs.user_cache_dir(appname="fsspec_dircache")
+            ) / str(listings_expiry_time)
+
+        try:
+            listings_cache_location.mkdir(exist_ok=True, parents=True)
+        except Exception:
+            logger.error(
+                f"folder for dircache could not be created at {listings_cache_location}"
+            )
+
+        self.cache_location = listings_cache_location
+
+        logger.info(f"Dircache located at {listings_cache_location}")
+
+        self._cache = Cache(directory=listings_cache_location)
+        self.use_listings_cache = use_listings_cache
```

Comment:
if use_listings_cache is False, should we end up here at all? Perhaps we need a NoopDirCache